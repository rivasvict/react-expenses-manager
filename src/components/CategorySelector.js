import React from 'react';

function CategorySelector({ handleChange, name, value, categoryOptions }) {
    return (
        <select name={name} value={value} onChange={handleChange}>
            <option value=''></option>
            {categoryOptions.map((categoryOption, key) => <option value={categoryOption.value} key={key}>{categoryOption.name}</option>)}
        </select>
    )
}

export default CategorySelector;