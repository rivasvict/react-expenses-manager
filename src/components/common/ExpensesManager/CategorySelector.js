import React from 'react';
import { FormSelect } from '../Forms';

function CategorySelector({ handleChange, name, value, categoryOptions, className = '' }) {
  return (
    <FormSelect className={className} name={name} value={value} onChange={handleChange}>
      <option value=''>Select a category</option>
      { /** TODO: Revise the set up of the categoryOption.value at this point
       * Probably a good idea not to make the frontend take care of the logic of the ,, */ }
      {categoryOptions.map((categoryOption, key) => <option value={`,${categoryOption.value},`} key={key}>{categoryOption.name}</option>)}
    </FormSelect>
  );
};

export default CategorySelector;
