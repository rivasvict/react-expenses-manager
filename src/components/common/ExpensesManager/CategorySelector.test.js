import React from "react";
import CategorySelector from "./CategorySelector";
import renderer from "react-test-renderer";

it("Renders CategorySelector component properly", () => {
  const tree = renderer
    .create(
      <CategorySelector
        handleChange={jest.fn()}
        name="CategorySelector"
        value=""
        categoryOptions={[
          { name: "Cat1", value: "cat1" },
          { name: "Cat2", value: "cat2" },
        ]}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
