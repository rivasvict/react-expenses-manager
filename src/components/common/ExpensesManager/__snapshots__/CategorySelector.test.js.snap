// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CategorySelector component properly 1`] = `
<select
  name="CategorySelector"
  onChange={[MockFunction]}
  value=""
>
  <option
    value=""
  />
  <option
    value="cat1"
  >
    Cat1
  </option>
  <option
    value="cat2"
  >
    Cat2
  </option>
</select>
`;
